src/main.rs

```diff
@@ -1,24 +1,47 @@
 mod config;
-use actix_web::{get, App, HttpServer, Responder};
+use actix_web::{web, App, HttpServer, Responder};
 use anyhow::Result as AnyhowResult;
 use tracing_actix_web::TracingLogger;
 
-#[get("/")]
-async fn root() -> impl Responder {
+async fn api_root() -> impl Responder {
     "I am root"
 }
 
+fn configure_api_scope(cfg: &mut web::ServiceConfig) {
+    cfg.service(web::resource("/").route(web::get().to(api_root)));
+}
+
 #[tokio::main]
 async fn main() -> AnyhowResult<()> {
     let _guard = config::logging::configure_tracing()?;
 
-    HttpServer::new(|| App::new().wrap(TracingLogger::default()).service(root))
-        .bind((
-            config::server::get_host().as_ref(),
-            config::server::get_port(),
-        ))?
-        .run()
-        .await?;
+    HttpServer::new(|| {
+        App::new().service(
+            web::scope("api/v1")
+                .wrap(TracingLogger::default())
+                .configure(configure_api_scope),
```

Comment:
With this change the HTTP API is registered as a "service" to our application. Later on we can go in and add a websocket "service" in a similar way.

I think having these things composed as different services in the app this way will be a good compromise with regards to separating responsibilities but not going through a lot of overhead before it's needed (like we might do if we wanted to split it out into it's own standalone server or something)

Reply:
Cool sounds good to me. One basic Rust organization question - if you were to add the websocket service in a similar way, would you just chain that along inside this same `main` function call, for example on line 25 here? Or would you try to abstract it to its own file or method?

Reply:
Yeah you'd want to add it as a new service to the application somehow and you'd definitely want to attach it in this same file, but the actual code that configures the websocket service can live wherever. I'll probably start by keeping it in this file for simplicity, but as it grows we'll probably want to move it to it's own module with it's own tests and whatnot